packages/eslint-plugin/src/rules/padding-line-between-statements.ts

```diff
@@ -0,0 +1,35 @@
+import baseRule from 'eslint/lib/rules/padding-line-between-statements';
+import {
+  InferOptionsTypeFromRule,
+  InferMessageIdsTypeFromRule,
+  createRule,
+} from '../util';
+
+export type Options = InferOptionsTypeFromRule<typeof baseRule>;
+export type MessageIds = InferMessageIdsTypeFromRule<typeof baseRule>;
+
+export default createRule<Options, MessageIds>({
+  name: 'padding-line-between-statements',
+  meta: {
+    type: 'layout',
+    docs: {
+      description: 'require or disallow padding lines between statements',
+      category: 'Stylistic Issues',
+      recommended: false,
+      extendsBaseRule: true,
+    },
+    fixable: 'whitespace',
+    schema: baseRule.meta.schema,
+    messages: {
+      unexpectedBlankLine: 'Unexpected blank line before this statement.',
+      expectedBlankLine: 'Expected blank line before this statement',
+    },
+  },
+  defaultOptions: [{ blankLine: 'any', prev: '*', next: '*' }],
+  create(context) {
+    const rules = baseRule.create(context);
+    return {
+      ...rules,
+    };
+  },
```

Comment:
Is there supposed to be some logic here?
Why is this extension rule doing literally nothing new?

Reply:
Sorry - I guess, I should have tried to capture the decisions a little more in the issue description.

So the initial suggestion was to add support for typescript constructs like `interface` and `type`. But this currently seems verify difficult to do since the `StatmentType` object is private to the rule and it is this object that is internally used by eslint to validate the code. 
Reference to [code](https://github.com/eslint/eslint/blob/2124e1b5dad30a905dc26bde9da472bf622d3f50/lib/rules/padding-line-between-statements.js#L350)

Now, the first attempt at the issue realized this and raised a PR with the ESLint team to modify the structure. The response from the ESLint was that they have frozen their stylistic rules completely.
Reference to [conversation](https://github.com/eslint/eslint/pull/13925#issuecomment-744107939)

The suggestion from them is to implement a custom rule or a plugin. The author for the first attempt created a plugin.

We now have 2 options - to support syntax either custom rule or an extension in the `typescript-eslint` repo.

I think my mistake was thinking to use the rule as-is we needed to route the rule through `typescript-eslint`. 
Which after verification doesn't seem to be the case.
Again sorry for wasting your time. 😞 

Let me know if you think we can probably support TS like syntax of the rule ? If not we should just close the issue

Reply:
There's no point creating an extension rule like this if we don't do anything at all to the rule!

If we can't do a simple (hacky) extension rule, then we likely need to fork the rule completely.
This is completely valid and okay.

We OFC prefer to avoid it if we can, but there are some rules that have hidden internal logic that we need to access and can't do so without forking.

So yes - feel free to copy whatever bits of the existing rule's code that we need to support TS language features.

Reply:
Forked the rule and added specific functionality related to TS.
Have tried to maintain it as close as possible to the original rule so that it is easier to review and maintain